### PR TITLE
Completed the predicate class

### PR DESCRIPTION
**Pull request by braunsonm**
This also adds improvements for the build so we can get proper test coverage

### CODE REVIEW

**Comment by braunsonm** [comment]
Don’t worry about all the commits, I’ll squash later.

**Comment by braunsonm** [comment]
Ok we’re finally good.

/cc @eileenvanheerde @mkoma808

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1,2 @@
 /src/ @mkoma808
+/src/__tets__/ @eileenvanheerde
```
Comment (eileenvanheerde) [inline .github/CODEOWNERS]: should be /src/__tests__/

**Comment by mkoma808** [comment]
Looks good.

**Pull request merged**